src/Apps/Artwork/Components/ArtistInfo.tsx

```diff
@@ -1,62 +1,156 @@
-import { Spacer, StackableBorderBox } from "@artsy/palette"
-import { artistResponse } from "Apps/__test__/Fixtures/MarketInsights"
-import React from "react"
+import React, { SFC } from "react"
+
+import { Box, Sans, Spacer, Spinner, StackableBorderBox } from "@artsy/palette"
+import { ArtistInfo_artist } from "__generated__/ArtistInfo_artist.graphql"
+import { ArtistInfoQuery } from "__generated__/ArtistInfoQuery.graphql"
+import { ContextConsumer } from "Artsy"
+import { Mediator } from "Artsy/SystemContext"
+import FollowArtistButton from "Components/FollowButton/FollowArtistButton"
+import { createFragmentContainer, graphql, QueryRenderer } from "react-relay"
+import { EntityHeader } from "Styleguide/Components/EntityHeader"
+import { get } from "Utils/get"
 
 import {
-  ArtistBio,
-  MarketInsights,
-  MarketInsightsProps,
-  SelectedExhibitions,
-  SelectedExhibitionsProps,
+  ArtistBioFragmentContainer as ArtistBio,
+  MarketInsightsFragmentContainer as MarketInsights,
+  SelectedExhibitionFragmentContainer as SelectedExhibitions,
 } from "Styleguide/Components"
-import { EntityHeader } from "Styleguide/Components/EntityHeader"
 
-interface ArtistInfoProps
-  extends MarketInsightsProps,
-    SelectedExhibitionsProps {
-  name: string
-  bio: string
-  credit?: string
+interface ArtistInfoProps {
+  artist: ArtistInfo_artist
+  user: User
+  mediator?: Mediator
 }
 
-export class ArtistInfo extends React.Component<ArtistInfoProps> {
-  render() {
-    // FIXME: Fetch from relay
-    const props = {
-      href: "/artist/francesca-dimattio",
-      imageUrl: "https://picsum.photos/110/110/?random",
-      name: "Francesca DiMattio",
-      meta: "American, b. 1979",
+export const ArtistInfo: SFC<ArtistInfoProps> = props => {
+  const imageUrl = get(props, p => p.artist.image.cropped.url)
+
+  return (
+    <>
+      <StackableBorderBox p={2} flexDirection="column">
+        <EntityHeader
+          name={props.artist.name}
+          meta={props.artist.formatted_nationality_and_birthday}
+          imageUrl={imageUrl}
+          FollowButton={
+            <FollowArtistButton
+              artist={props.artist}
+              user={props.user}
+              onOpenAuthModal={() => {
+                props.mediator.trigger("open:auth", {
+                  mode: "signup",
+                  copy: `Sign up to follow ${props.artist.name}`,
+                  signupIntent: "follow artist",
+                  afterSignUpAction: {
+                    kind: "artist",
+                    action: "follow",
+                    objectId: props.artist.id,
+                  },
+                })
+              }}
+              render={({ is_followed }) => {
```

Comment:
Surprised that this render is inline here and not part of `FollowArtistButton` component.

Reply:
This is the `renderProps` pattern, something I updated to the Follow button a while back. Provide a render prop that's a function and call from inside the component's render function. What is returned from it [then overwrites the the look and feel](https://github.com/artsy/reaction/blob/master/src/Components/FollowButton/FollowArtistButton.tsx#L124-L126) of the button itself. This is a common pattern for customizing things; in this case its so that we can easily use our new design system from within the button.

Reply:
I see. So we can support both button and follow text. Nice:)